File: leetcode_problems/cpp/lc682_baseball_game.cpp
#include <vector>
#include <string>
#include <stack>
using std::stack;
using std::string;
using std::vector;

class Solution
{
public:
    int calPoints(vector<string> &operations)
    {
        stack<int> st;
        for (string op : operations)
        {
            if (op == "+")
            {
                int t1 = st.top();
                st.pop();
                int t2 = st.top();
                st.push(t1);
                st.push(t1 + t2);
            }
            else if (op == "D")
            {
                st.push(st.top() * 2);
            }
            else if (op == "C")
            {
                st.pop();
            }
            else
            {
                st.push(std::stoi(op));
            }
        }

        int sum = 0;
        while (!st.empty())
        {
            sum += st.top();
            st.pop();
        }
        return sum;
    }
};

File: leetcode_problems/typescript/src/lc682_baseball_game.ts
function calPoints(operations: string[]): number {
  let stack: number[] = [];
  for (let op of operations) {
    if (op == "+") {
      stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
    } else if (op == "D") {
      stack.push(stack[stack.length - 1] * 2);
    } else if (op == "C") {
      stack.pop();
    } else {
      stack.push(parseInt(op));
    }
  }
  let sum = 0;
  for (let num of stack) {
    sum += num;
  }
  return sum;
}
